sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileIndex.scala

```diff
@@ -72,14 +72,4 @@ trait FileIndex {
 
   /** Schema of the partitioning columns, or the empty schema if the table is not partitioned. */
   def partitionSchema: StructType
-
-  /**
-   * Returns an optional metadata operation time, in nanoseconds, for listing files.
-   *
-   * We do file listing in query optimization (in order to get the proper statistics) and we want
-   * to account for file listing time in physical execution (as metrics). To do that, we save the
-   * file listing time in some implementations and physical execution calls it in this method
-   * to update the metrics.
-   */
-  def metadataOpsTimeNs: Option[Long] = None
```

Comment:
This is part of https://github.com/apache/spark/pull/23298/commits/73bf55389466a553606d1878848c0bb018b36dc2, here I revert the whole changes in [SPARK-20151](https://github.com/apache/spark/pull/17476/files). Maybe use the QueryPlanningTracker to measure the pruning phase will have a better code base in `FileSourceScanExec`.